pkg/console/subresource/configmap/configmap.go

```diff
@@ -40,54 +32,62 @@ func getApiUrl(infrastructureConfig *configv1.Infrastructure) string {
-	host := rt.Spec.Host
-	apiServerURL := getApiUrl(infrastructureConfig)
-	defaultProviders := operatorv1.ConsoleProviders{}
-	defaultConfig := NewYamlConfig(host, defaultLogoutURL, DEFAULT_BRAND, DEFAULT_DOC_URL, apiServerURL, defaultProviders)
-	// Get console config from the openshift-config-managed namespace
+func statusPageId(operatorConfig *operatorv1.Console) string {
+	if operatorConfig.Spec.Providers.Statuspage != nil {
+		return operatorConfig.Spec.Providers.Statuspage.PageID
+	}
+	return ""
+}
+
+func DefaultConfigMap(
+	operatorConfig *operatorv1.Console,
+	consoleConfig *configv1.Console,
+	managedConfig *corev1.ConfigMap,
+	infrastructureConfig *configv1.Infrastructure,
+	rt *routev1.Route) (consoleConfigmap *corev1.ConfigMap, unsupportedConfigMerged bool, err error) {
+
+	defaultBuilder := &consoleserver.ConsoleServerCLIConfigBuilder{}
+	defaultConfig, err := defaultBuilder.Host(rt.Spec.Host).
+		LogoutURL(defaultLogoutURL).
+		Brand(DEFAULT_BRAND).
+		DocURL(DEFAULT_DOC_URL).
+		APIServerURL(getApiUrl(infrastructureConfig)).
+		ConfigYAML()
+
 	extractedManagedConfig := extractYAML(managedConfig)
 
-	// Derive a user defined config using operator config
-	logoutRedirect := consoleConfig.Spec.Authentication.LogoutRedirect
-	brand := operatorConfig.Spec.Customization.Brand
-	docURL := operatorConfig.Spec.Customization.DocumentationBaseURL
-	providers := operatorConfig.Spec.Providers
-	userDefinedConfig := NewYamlConfig(host, logoutRedirect, brand, docURL, apiServerURL, providers)
+	userDefinedBuilder := &consoleserver.ConsoleServerCLIConfigBuilder{}
+	userDefinedConfig, err := userDefinedBuilder.Host(rt.Spec.Host).
+		LogoutURL(consoleConfig.Spec.Authentication.LogoutRedirect).
+		Brand(operatorConfig.Spec.Customization.Brand).
+		DocURL(operatorConfig.Spec.Customization.DocumentationBaseURL).
+		APIServerURL(getApiUrl(infrastructureConfig)).
+		StatusPageID(statusPageId(operatorConfig)).
+		ConfigYAML()
 
 	unsupportedConfigOverride := operatorConfig.Spec.UnsupportedConfigOverrides.Raw
-
-	// merge configs with overrides, if we have them
-	mergedConfig, err := resourcemerge.MergeProcessConfig(nil, defaultConfig, extractedManagedConfig, userDefinedConfig, unsupportedConfigOverride)
-	if err != nil {
-		klog.Errorf("failed to merge configmap: %v", err)
-		return nil, false, err
+	willMergeConfigOverrides := len(unsupportedConfigOverride) != 0
+	if willMergeConfigOverrides {
+		klog.V(4).Infoln(fmt.Sprintf("with UnsupportedConfigOverrides: %v", string(unsupportedConfigOverride)))
 	}
 
-	outConfigYaml, err := yaml2.JSONToYAML(mergedConfig)
+	merger := &consoleserver.ConsoleYAMLMerger{}
+	mergedConfig, err := merger.Merge(
+		defaultConfig,
+		extractedManagedConfig,
+		userDefinedConfig,
+		operatorConfig.Spec.UnsupportedConfigOverrides.Raw)
 	if err != nil {
-		klog.Errorf("failed to generate configmap: %v", err)
+		klog.Errorf("failed to generate configmap: %v \n", err)
```

Comment:
no need to add... `klog` will deal with it for us :)
```suggestion
		klog.Errorf("failed to generate configmap: %v", err)
```

Reply:
Ah forgot to update.

Reply:
fixeed.